dev/tests/integration/framework/Magento/TestFramework/Annotation/DbIsolation.php

```diff
@@ -49,6 +131,20 @@ public function endTestTransactionRequest(
     ) {
         if ($this->_isIsolationActive && $this->_getIsolation($test)) {
             $param->requestTransactionRollback();
+        } else {
+            $isolationProblem = [];
+            foreach ($this->dbStateTables as $dbStateTable => $method) {
+                $data = $this->pullDbState($dbStateTable);
+                $data = $this->{$method}($data);
+
+                if ($data) {
+                    $isolationProblem[$dbStateTable] = $data;
+                }
+            }
+
+            if (!empty($isolationProblem)) {
+                throw new \Exception("There was a problem with isolation: " . var_export($isolationProblem, true));
```

Comment:
we can't throw an exception due to this will lead to un-readable error present only in logs:

```
[Integration-14-ce] {"messages":{"error":[{"code":500,"message":"Fatal Error: 'Uncaught Exception: There was a problem with isolation: array (\n  'catalog_product_entity' => \n  array (\n    0 => \n    array (\n      'entity_id' => '1',\n      'attribute_set_id' => '4',\n      'type_id' => 'simple',\n      'sku' => 'simple',\n      'has_options' => '1',\n      'required_options' => '1',\n      'created_at' => '2020-08-14 13:33:42',\n      'updated_at' => '2020-08-14 13:33:42',\n    ),\n  ),\n) in \/var\/www\/html\/dev\/tests\/integration\/framework\/Magento\/TestFramework\/Annotation\/DbIsolation.php:146\nStack trace:\n#0 [internal function]: Magento\\TestFramework\\Annotation\\DbIsolation->endTestTransactionRequest()\n#1 \/var\/www\/html\/dev\/tests\/integration\/framework\/Magento\/TestFramework\/EventManager.php(49): call_user_func_array()\n#2 \/var\/www\/html\/dev\/tests\/integration\/framework\/Magento\/TestFramework\/Event\/Transaction.php(76): Magento\\TestFramework\\EventManager->fireEvent()\n#3 \/var\/www\/html\/dev\/tests\/integration\/framework\/Magento\/TestFramework\/Event\/Transaction.php(56): Magento\\TestFramework\\Event\\Transaction->_proc' in '\/var\/www\/html\/dev\/tests\/integration\/framework\/Magento\/TestFramework\/Annotation\/DbIsolation.php' on line 146","trace":"Trace is not available."}]}}[Pipeline] [Integration-6-ce] sh
[Integration-6-ce] + docker exec 81d7f2dcf095739d911ecd8e192aa60c549fbaf64852ee8d43972973b2bc69b3 /bin/bash -c 'sudo -E -u magento cp /var/www/html/dev/build/core_dev/integration/search-engine-config-elasticsearch7.xml /var/www/html/dev/tests/integration/_files/Magento/TestModuleCatalogSearch/etc/config.xml || true; '

```